torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -1293,10 +1293,18 @@ def _post_backward_hook(self, param: Parameter, *unused: Any) -> None:
                 # Average grad by world_size for consistency with PyTorch DDP.
                 param.grad.div_(self.gradient_predivide_factor)
 
+            grad = param.grad.data
             if param._is_sharded:  # type: ignore[attr-defined]
-                grad_flatten = torch.flatten(param.grad)
+                # We clear `param.grad` to permit repeated gradient
+                # computations when this FSDP module is called multiple times.
```

Comment:
could we add a comment to explain why the race could happen if not clear param.grad. "This is to avoid race among multiple re-entrant backward passes. for multiple re-entrant backward passes, e.g., it is possible that the second backward pass computation is ahead of first backward pass reduction, as reduction is in another stream and is async. In this case, first backward pass may be reducing the second pass's param.grad, which is not correct."